packages/drawnix/src/components/toolbar/popup-toolbar/line-arrow-button.tsx

```diff
@@ -0,0 +1,169 @@
+  const container = PlaitBoard.getBoardContainer(board);
+
+  const setSourceMarker = (marker: string) => {
+    PropertyTransforms.setProperty(board, {
+      source: {
+        ...source,
+        marker,
+      },
+    });
+  };
+
+  const setTargetMarker = (marker: string) => {
+    PropertyTransforms.setProperty(board, {
+      target: {
+        ...target,
+        marker,
+      },
+    });
+  };
+
+  return (
+    <>
+      <Popover
+        sideOffset={12}
+        open={isSourceOpen}
+        onOpenChange={(open) => {
+          setIsSourceOpen(open);
+        }}
+        placement={'top'}
+      >
+        <PopoverTrigger asChild>
+          <ToolButton
+            className={classNames(`property-button  source-arrow-button`)}
+            visible={true}
+            icon={sourceMarker === 'none' ? LineIcon : ArrowIcon}
+            type="button"
+            title={title}
+            aria-label={title}
+            onPointerUp={() => {
+              setIsSourceOpen(!isSourceOpen);
+            }}
+          ></ToolButton>
+        </PopoverTrigger>
+        <PopoverContent container={container}>
+          <Island
+            padding={2}
+            className={classNames(`${ATTACHED_ELEMENT_CLASS_NAME} source-arrow-island`)}
+          >
+            <Stack.Row gap={1}>
+              <ToolButton
+                className={classNames(`property-button`)}
+                visible={true}
+                icon={LineIcon}
+                type="button"
+                title={title}
+                aria-label={title}
+                selected={sourceMarker === 'none'}
+                onPointerUp={() => {
+                  setSourceMarker('none');
+                }}
+              ></ToolButton>
+              <ToolButton
+                className={classNames(`property-button`)}
+                visible={true}
+                icon={ArrowIcon}
+                type="button"
+                title={title}
+                selected={sourceMarker === 'arrow'}
+                aria-label={title}
+                onPointerUp={() => {
+                  setSourceMarker('arrow');
+                }}
+              ></ToolButton>
+            </Stack.Row>
+          </Island>
+        </PopoverContent>
+      </Popover>
+
+      <Popover
+        sideOffset={12}
+        open={isTargetOpen}
+        onOpenChange={(open) => {
+          setIsTargetOpen(open);
+        }}
+        placement={'top'}
+      >
+        <PopoverTrigger asChild>
+          <ToolButton
+            className={classNames(`property-button`)}
+            visible={true}
+            icon={targetMarker === 'none' ? LineIcon : ArrowIcon}
+            type="button"
+            title={title}
+            aria-label={title}
+            onPointerUp={() => {
+              setIsTargetOpen(!isTargetOpen);
+            }}
+          ></ToolButton>
+        </PopoverTrigger>
+        <PopoverContent container={container}>
+          <Island
```

Comment:
My suggestions in component struct:

1. Moving the segment to the individual component  named as `arrow-mark-picker`.
2. We could name the current component as `arrow-mark-button` and only contain one arrow mark setting and it will be used twice

Do you think it's better than now?